File: CsExample/EndpointDefinitions/TestEndpointDefinition.cs
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace CsExample.EndpointDefinitions
{
    public class TestEndpointDefinition : IEndpointDefinition
    {
        public void DefineServices(WebApplicationBuilder appBuilder) { }

        public void DefineEndpoints(WebApplication app)
        {
            static string SayHello() => "Hello";
            static string SayHelloTo([FromBody] string name) => "Hello, " + name;

            app.MapGet("/Test", SayHello).WithTags("Test");
            app.MapPost("/Test", SayHelloTo).WithTags("Test");
        }
    }
}

File: FsExample/EndpointDefinitions/TestEndpointDefinition.fs
module TestEndpointDefinition

open System

open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Http


[<CLIMutable>]
type TestParams = {
    Name: string
}

[<Sealed>]
type TestEndpointDefinitions() =

    let sayHello () = "Hello"
    let sayHelloToString name = sprintf "Hello, %s" name
    let sayHelloTo testParams = sayHelloToString testParams.Name

    interface IEndpointDefinition with
        member _.DefineServices(_) =
            ()

        member this.DefineEndpoints(app) =
            app.MapGet("/Test", Func<_> sayHello).WithTags("Test") |> ignore
            app.MapPost("/Test", Func<_> sayHello).WithTags("Test") |> ignore

            app.MapGet("/TestWithQuery", Func<_,_> (fun name -> sayHelloToString name)).WithTags("TestWithQuery") |> ignore
            app.MapPost("/TestWithQuery", Func<_,_> (fun name -> sayHelloToString name)).WithTags("TestWithQuery") |> ignore

            app.MapPost("/TestWithBody", Func<_,_> sayHelloTo).WithTags("TestWithBody") |> ignore
